src/sc_kpm/utils/action_utils.py

```diff
@@ -120,9 +117,10 @@ def finish_action(action_node: ScAddr, status: QuestionStatus = QuestionStatus.Q
 
 
 def finish_action_with_status(action_node: ScAddr, is_success: bool = True) -> None:
-    if is_success:
-        status_node = QuestionStatus.QUESTION_FINISHED_SUCCESSFULLY
-    else:
-        status_node = QuestionStatus.QUESTION_FINISHED_UNSUCCESSFULLY
-    finish_action(action_node, status_node)
     finish_action(action_node, QuestionStatus.QUESTION_FINISHED)
+    finish_action(
+        action_node,
+        QuestionStatus.QUESTION_FINISHED_SUCCESSFULLY
+        if is_success
+        else QuestionStatus.QUESTION_FINISHED_UNSUCCESSFULLY,
```

Comment:
For me, this is hard-to-read function calling. Looks like function takes 4 args, not 2